### PR TITLE
[Android] invalidate shadows on parent's size change

### LINKED ISSUE #24034: Shadow is not updating on change of parent control.
### Description

Hey,
I am facing an issue in Shadow.
When I changed the size of Shadow's parent control, I expected Shadow to update it.
It's working fine with iOS but not working in Android.
I tried with different parent controls like Border, VerticalStackLayout, etc. but the results are the same.
Can someone look into this?

App Code: 
[Test2.zip](https://github.com/user-attachments/files/16509881/Test2.zip)

Not working sample in Android:

https://github.com/user-attachments/assets/fc7c192b-31e4-4f88-81e6-ad8b808f409e

Working sample in iOS:

https://github.com/user-attachments/assets/35480344-c02c-4170-b583-8b27d9f6c4dc

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.70 SR7

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

All versions.

### Did you find any workaround?

No

### Relevant log output

_No response_

### LINKED ISSUE #24476: Changing view size/layout/position does not update the shadow
### Description

I have some code:

```xaml
<Border
    Grid.Row="1"
    Grid.ColumnSpan="2"
    VerticalOptions="End"
    Margin="0,0,0,24"
    HeightRequest="52"
    StrokeThickness="0"
    Background="#EEEEEE"
    StrokeShape="RoundRectangle 26"
    MaximumWidthRequest="300"
    HorizontalOptions="Center">
    <Border.Shadow>
        <Shadow
            Offset="0,25"
            Opacity="0.3"
            Radius="25"/>
    </Border.Shadow>
```

There are some items in the border, and if I change (for example) the HorizontalOptions to Fill or Center, the shadow just moves but n=does not reflect the bounds.

This is noticeable during hot reloading. 

However, this appears to be working fine if I change the contents - such as hid/show - or other content things.

### Steps to Reproduce

image

image

### Link to public reproduction project repository

https://github.com/mattleibow/scorecard-day-nft-marketplace/tree/issues/shadow

### Version with bug

9.0.0-preview.7.24407.4

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/24034
Fixes https://github.com/dotnet/maui/issues/24476

https://github.com/user-attachments/assets/5474d4a4-5fea-4041-8e60-4a6eeed0db79

### CODE REVIEW

File: src/Core/src/Platform/Android/WrapperView.cs [hunk 1]
```diff
@@ -148,11 +151,13 @@ protected override void DrawShadow(Canvas canvas, int viewWidth, int viewHeight)
 			Graphics.Color solidColor = null;
 
 			// If need to redraw shadow
-			if (_invalidateShadow)
+			if (_invalidateShadow || DidViewSizeChange(viewWidth, viewHeight))
```
[the remark this answers is not in the document]
Comment (kubaflo) [reply src/Core/src/Platform/Android/WrapperView.cs]: Yes, adding `_invalidateShadow = true; ` to `OnLayout` is enough :)
I've pushed a commit

File: src/Core/src/Platform/Android/WrapperView.cs [hunk 2]
```diff
@@ -55,6 +55,7 @@ protected override void OnLayout(bool changed, int left, int top, int right, int
 			var widthMeasureSpec = MeasureSpecMode.Exactly.MakeMeasureSpec(right - left);
 			var heightMeasureSpec = MeasureSpecMode.Exactly.MakeMeasureSpec(bottom - top);
 
+			_invalidateShadow = true;
```
Comment (mattleibow) [inline src/Core/src/Platform/Android/WrapperView.cs:58]: Do we need to do this is the positions (X/Y) changes as well, or just size? I can't recall if the shadow is automatically positioned based on the view origin or also gets "stuck"

**Review by mattleibow — COMMENTED** [review]
I think this is good to go, just got a Q about the invalidation for any bounds change or just size.

File: src/Core/src/Platform/Android/WrapperView.cs (hunk 2 above)
Comment (mattleibow) [reply src/Core/src/Platform/Android/WrapperView.cs:58]: We can do this if it actually becomes a problem.

**Review by mattleibow — APPROVED** [review]

**Comment by PureWeen** [comment]
Failing iOS test is unrelated and was already fixed on main

**Pull request merged**